### PR TITLE
Message Sending User Experience in Lotus

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

File: proposals/15-message-sending-ux-in-lotus.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+<!--(short paragraph)-->
+
+_All of the below are reported issues with message sending UX._
+
+- Sending message from (partially) unsynced node, resulting in wrongly estimated gas fees.
+	- **Solution**: warn the user that node is not in sync, ask for confirmation to send the message
+- Default `MaxFee` configuration option results in message fees being too low, resulting in message not landing on chain:
+	- **Solution**: for interactive usage use the `MaxFee` as guideline. If message wouldn't immediately land on chain due to it (or would be at a risk of being squeezed out in short time), suggest to the user alternative fee. The alternative fee can be accepted, refused or modified. If low fee is specified, estimate the time message will need to land.
+- Previous messages are blocking new messages due to changing network conditions or insufficient funds.
+	- **Solution part 1**:  Warn users when new messages are submitted and ask for confirmation that user wants to proceed.
```
Comment (momack2) [inline proposals/15-message-sending-ux-in-lotus.md:45]: Seems like more so than “do you want to proceed” is an affordance to also resolve the blocker (other blocked messages in mpool) - in general, giving users an affordance to understand and resolve the underlying problem is more useful/actionable

File: proposals/15-message-sending-ux-in-lotus.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	- **Solution part 2**: Build an interactive tool for observing state of messages, allowing the user to reprice them, discard them and showing the user what is the exact issue with them (head of the line blocking, insufficient funds, too low fee)
+
+#### Impact
+<!--\_How directly important is the outcome to web3 dev stack product-market fit?\_-->
+
+🔥🔥 
+This project has potential to decrease attrition of new users that are starting to experiment with Filecoin, while at the same time improving the experience of existing users.
```
Comment (momack2) [inline proposals/15-message-sending-ux-in-lotus.md:52]: Is this only relevant for lotus nodes that have sync’d the whole chain or also folks who want to do more lightweight message sending?

File: proposals/15-message-sending-ux-in-lotus.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	- Clear indication of past messages, number of confirmations, exit status.
+	- Provide a way to resolve "struck" messages, interactive functionality of `mpool replace`.
+
+#### What does done look like?
+<!--\_What specific deliverables should completed to consider this project done?\_-->
+Issues identified in the first step are resolved, users are protected against issues mentioned in this document.
+
+####  What does success look like?
+<!--\_Success means impact. How will we know we did the right thing?\_-->
+Number of issues and complaints by users and end users regarding message related interactions with Filecoin is reduced. Message sending related problems are no longer mentioned when on-boarding new users.
```
Comment (momack2) [inline proposals/15-message-sending-ux-in-lotus.md:93]: Could we look at metrics like total messages on chain, # clients that have sent at least one message, etc?

**Review by momack2 — COMMENTED** [review]
This seems pretty important! @raulk @willscott is this on either of your radar?

**Comment by willscott** [comment]
> This seems pretty important! @raulk @willscott is this on either of your radar?

it looks like @Kubuxu is going to be working on at least some of this, right?

**Comment by raulk** [comment]
Yes, this is green lighted for Ignite, and we'll be starting on it soon 💪

File: proposals/15-message-sending-ux-in-lotus.md (hunk 1 above)
Comment (Kubuxu) [reply proposals/15-message-sending-ux-in-lotus.md:45]: Yeah, it should suggest resolution using the `Solution part 2`.

File: proposals/15-message-sending-ux-in-lotus.md (hunk 3 above)
Comment (Kubuxu) [reply proposals/15-message-sending-ux-in-lotus.md:93]: Total messages won't tell us anything as 99.9% of them are from automated systems.
We can maybe look at: "new first time message senders".

File: proposals/15-message-sending-ux-in-lotus.md (hunk 2 above)
Comment (Kubuxu) [reply proposals/15-message-sending-ux-in-lotus.md:52]: You don't have to sync the whole chain and also I would also imagine some of the features could be supported by `lotus daemon --lite` without any additional work, where the `--lite` doesn't require any syncing at all.

**Review by mikeal — APPROVED** [review]

**Comment by jacobheun** [comment]
Released in Lotus 1.11.0, merging

**Pull request merged**